### PR TITLE
Add SyncBatchNormalization layer for TensorFlow.

### LINKED ISSUE #2066: Sync Batch Norm for Tensorflow
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here.

Are there Sync Batch Norm implementations for tensorflow in horovod ?

### PR DESCRIPTION
**Pull request by romerojosh**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [x] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description
This PR adds a `SyncBatchNormalization` layer implementation using Horovod for TensorFlow.  

Fixes #2066.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by romerojosh** [comment]
@tgaddair As discussed, would we be open to removing support for the very old TF release in our tests (v1.6) to unblock this PR?

**Comment by tgaddair** [comment]
Sure, I can take a stab at that today.

File: horovod/tensorflow/sync_batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      kwargs['name'] = 'sync_batch_normalization'
+    super(SyncBatchNormalization, self).__init__(fused=fused, **kwargs)
+
+  def _moments(self, inputs, reduction_axes, keep_dims):
+    """Compute the mean and variance: it overrides the original _moments."""
+
+    worker_mean, worker_variance = super(SyncBatchNormalization, self)._moments(
+      inputs, reduction_axes, keep_dims=keep_dims)
+
+    if size() > 1:
```
Comment (tgaddair) [inline horovod/tensorflow/sync_batch_norm.py:42]: We may want to make this work with dynamic worker count in a follow-up PR.

Comment (romerojosh) [reply horovod/tensorflow/sync_batch_norm.py:42]: Yes, sounds good to me. Thanks for pointing that out and opening the issue.

**Comment by weiminggao** [comment]
I test batch norm in tf 1.14 with graph mode , it does not work. Why
